comp/core/flare/helpers/builder.go

```diff
@@ -106,6 +107,9 @@ func NewFlareBuilder(localFlare bool) (types.FlareBuilder, error) {
 
 // builder implements the FlareBuilder interface
 type builder struct {
+	sync.Mutex
+	done bool
```

Comment:
A few method method are still accessing the internal data without checking `done`: `logError`, `logf`, `PrepareFilePath`, `RegisterFilePerm`, `RegisterDirPerm`, ...

Could we create a helper function called everywhere to check `done`. We might also want to check for as early as possible to avoid non-necessary scrubbing.

Also, what do you think about `isClosed` instead of `done` ?

Reply:
I only added `done` to avoid returning an error on write when the flare was already created, it's not an issue for those functions since they don't return an error*

_*actually `Logf` returns an error if it fails to write in the log file so I should add a check in that one too_

Checking early if the flare is already done is a good idea, I'll do that 👍

Reply:
I only added `done` to avoid returning an error on write when the flare was already created, it's not an issue for those functions since they don't return an error*

_*actually `Logf` returns an error if it fails to write in the log file so I should add a check in that one too, my bad_

Checking early if the flare is already done is a good idea, I'll do that 👍